static/js/reactions.js

```diff
@@ -523,10 +526,24 @@ function make_clean_reaction({local_id, user_ids, emoji_name, emoji_code, reacti
 
 export function update_user_fields(r) {
     r.count = r.user_ids.length;
+    r.names_reacted = get_names_reacted(r.user_ids);
     r.label = generate_title(r.emoji_name, r.user_ids);
     if (r.user_ids.includes(page_params.user_id)) {
         r.class = "message_reaction reacted";
     } else {
         r.class = "message_reaction";
     }
 }
+
+// helper function for setting name/count for a reaction
+function get_names_reacted(user_list) {
+    const usernames = people.get_display_full_names(user_list);
+    if (user_list.length === 1) {
+        return usernames[0];
+    } else if (user_list.length === 2) {
+        const names_reacted = `${usernames[0]} and 1 other`;
+        return names_reacted;
+    }
+    const names_reacted = `${usernames[0]} and ${user_list.length - 1} others`;
+    return names_reacted;
```

Comment:
I think we were wanting to just join names with `,`, right?  That will fix the lack of internationalization in this PR naturally.